### PR TITLE
Validation draft

### PR DESCRIPTION
**Pull request by maurafortino**

### CODE REVIEW

File: options.go [hunk 1]
```diff
@@ -31,47 +31,53 @@ func (e errorOption) String() string {
...
 
-// AtLeastOneEvent makes sure there is at least one value in Events and ensures
-// that all values should parse into regex.
-func AtLeastOneEvent() Option {
-	return atLeastOneEventOption{}
+func AlwaysValid() Option {
+	return AlwaysValidOption{}
 }
 
-type atLeastOneEventOption struct{}
```
Comment (denopink) [inline options.go:40]: is `atLeastOneEventOption` used by anything currently in production?

File: options.go [hunk 2]
```diff
@@ -31,47 +31,53 @@ func (e errorOption) String() string {
...
 }
 
-// EventRegexMustCompile ensures that all values in Events parse into valid regex.
-func EventRegexMustCompile() Option {
-	return eventRegexMustCompileOption{}
+func ValidateEvents() Option {
+	return validateEvents{}
 }
 
-type eventRegexMustCompileOption struct{}
```
Comment (denopink) [inline options.go:59]: is `eventRegexMustCompileOption` currently used in production?

File: options.go [hunk 3]
```diff
@@ -171,93 +152,39 @@ func (p provideTimeNowFuncOption) String() string {
...
-		}
+	err := r.ValidateURLs(p.checker)
+	if err != nil {
+		return fmt.Errorf("%w:%w", ErrInvalidInput, err)
 	}
+
 	return nil
 }
 
-func (p provideReceiverURLValidatorOption) String() string {
```
Comment (denopink) [inline options.go:225]: is `ProvideReceiverURLValidator` currently used in production?

File: options.go [hunk 4]
```diff
@@ -171,93 +152,39 @@ func (p provideTimeNowFuncOption) String() string {
...
-		return "ProvideReceiverURLValidator(nil)"
-	}
-	return "ProvideReceiverURLValidator(" + p.checker.String() + ")"
-}
+// ProvideReceiverURLValidator is an option that allows the caller to provide a
+// URL validator that is used to validate the ReceiverURL.
 
 // ProvideAlternativeURLValidator is an option that allows the caller to provide
 // a URL validator that is used to validate the AlternativeURL.
-func ProvideAlternativeURLValidator(checker *urlegit.Checker) Option {
```
Comment (denopink) [inline options.go:234]: is `ProvideAlternativeURLValidator` currently used in production?

File: options.go [hunk 5]
```diff
@@ -157,8 +138,8 @@ type provideTimeNowFuncOption struct {
 	nowFunc func() time.Time
 }
 
-func (p provideTimeNowFuncOption) Validate(r *Registration) error {
-	r.nowFunc = p.nowFunc
+func (p provideTimeNowFuncOption) Validate(r Register) error {
+	// r.nowFunc = p.nowFunc
```
Comment (denopink) [inline options.go:142]: is this needed or not?

**Review by denopink — APPROVED** [review]
a few questions, otherwise looks good

**Comment by maurafortino** [comment]
closing without merging - as https://github.com/xmidt-org/webhook-schema/pull/16 has been merged and issues addressed here were fixed/updated in that MR

**Pull request closed without merging**